src/validation.cpp

```diff
@@ -4390,143 +4390,22 @@ bool CChainState::ReplayBlocks(const CChainParams& params)
     return true;
 }
 
-//! Helper for CChainState::RewindBlockIndex
-void CChainState::EraseBlockData(CBlockIndex* index)
+bool CChainState::NeedsIBD(const CChainParams& params)
```

Comment:
Consider renaming this function `NeedsRedownload`. All nodes that are starting fresh or restarting after some time need IBD. This function is specifically for nodes that need to delete blocks and redownload them.

Reply:
Makes sense. Done.